src/mobu/services/github_ci/ci_manager.py

```diff
@@ -0,0 +1,337 @@
+"""Manager for background workers that process work from GitHub checks."""
+
+from __future__ import annotations
+
+import asyncio
+from asyncio import Queue
+from dataclasses import dataclass, field
+from typing import TypeAlias
+
+from aiojobs import Job, Scheduler
+from httpx import AsyncClient
+from safir.github import GitHubAppClientFactory
+from structlog.stdlib import BoundLogger
+
+from ...config import config
+from ...models.ci_manager import CiManagerSummary, CiWorkerSummary
+from ...models.user import User
+from ...storage.gafaelfawr import GafaelfawrStorage
+from ...storage.github import GitHubStorage
+from .ci_notebook_job import CiNotebookJob
+
+__all__ = ["CiManager"]
+
+
+@dataclass
+class CiManagerLifecycle:
+    marked_remaining: asyncio.Event = field(default_factory=asyncio.Event)
+
+
+@dataclass
+class JobLifecycle:
+    processing: asyncio.Event = field(default_factory=asyncio.Event)
+    processed: asyncio.Event = field(default_factory=asyncio.Event)
+
+
+@dataclass
+class QueuedJob:
+    job: CiNotebookJob
+    lifecycle: JobLifecycle
+
+
+QueueItem: TypeAlias = QueuedJob | None
```

Comment:
Why can a `QueueItem` be `None`? What does pushing `None` onto the queue mean?

Reply:
`None` is a magic value to make the workers stop listening on the queue.  I changed this to be an explicit `"stop"` literal.